File: moodfm-backend/src/main/java/com/moodfm/domain/vo/SongVO.java
package com.moodfm.domain.vo;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class SongVO {
    private Long id;
    private String title;
    private String artist;
    private String album;
    private Integer durationSeconds;
    private String coverUrl;
    private String platform;
    private String platformSongId;
    private String playUrl;
    private String recommendReason; // AI 推荐解释
    private String urlSource; // "qqmusic" | "netease_fallback" — set when play URL comes from a different platform
    private String features;  // JSON string from Song.features; null if not yet enriched
}

File: moodfm-frontend/src/utils/songVo.ts
import type { Song } from '@/types'

/** Map backend SongVO (radio/search responses) to player Song. */
export function mapSongVoToSong(s: {
  id?: number | string
  title: string
  artist: string
  album?: string
  platform?: string
  platformSongId?: string
  durationSeconds?: number
  duration?: number
  coverUrl?: string
  playUrl?: string
  recommendReason?: string
  urlSource?: string
}): Song {
  return {
    id: String(s.id ?? ''),
    title: s.title,
    artist: s.artist,
    album: s.album ?? '',
    platform: (s.platform as Song['platform']) || undefined,
    platformSongId: s.platformSongId ?? '',
    duration: s.durationSeconds ?? s.duration ?? 0,
    coverUrl: s.coverUrl ?? '',
    audioUrl: s.playUrl ?? undefined,
    recommendReason: s.recommendReason,
    urlSource: s.urlSource,
  }
}